neuralmonkey/decoding_function.py

```diff
@@ -178,9 +179,10 @@ def attention(self, decoder_state: tf.Tensor,
             if self.input_weights is None:
                 weights = tf.nn.softmax(s)
             else:
-                weights_all = tf.nn.softmax(s) * self.input_weights
-                norm = tf.reduce_sum(weights_all, 1, keep_dims=True) + 1e-8
-                weights = weights_all / norm
+                masked_logits = tf.where(
+                    tf.equal(self.input_weights, 1),
+                    s, -np.inf * tf.ones_like(s))
```

Comment:
zkus sem dát místo toho ones_like tohle:

```
tf.fill(dims=tf.shape(s), value=-np.inf)
```

a mohlo by to i broadcastovat

Reply:
proč to nefunguje? co to píše?

Reply:
porad stejnou hlasku ... mam totiz pocit ze si to nestezuje na ty np.inf ale na to kdyz to vraci pouze s, coz je ale divne

Reply:
Stěžuje si operace `Select`, což je jméno pro to volání toho `where`, viz TF kód. Stěžuje si, že `s` a `-np.inf * tf.ones_like(s)` mají různý shape. `ones_like(s)`by mělo vracet shape jako má `s`. Takže to obal printama a zjisti, jaký shape mají tenzory `s`, `xxx = tf.ones_like(s)` a `-np.inf * xxx` a snad ti to dá nějakej insight.  Rovnou zjisti i jakej shape má ten `tf.equal(...)` - ten where se chová jinak, když má condition jednu dimenzi a když jich má víc.

Bude tam demence v tom, že v prvnim kroku beam search je batch size jedna (vycházim ze stejnýho stavu a se stejným vstupem a vemu beam nejlepších) a v ostatních krocích je batch size velikost beamu.